src/pydantic_zarr/v2.py

```diff
@@ -1045,3 +1059,55 @@ def auto_dimension_separator(data: Any) -> Literal["/", "."]:
     if hasattr(data, "dimension_separator"):
         return data.dimension_separator
     return "/"
+
+
+def _guess_chunks(shape: tuple[int, ...], typesize: int) -> tuple[int, ...]:
```

Comment:
we have an [equivalent](https://github.com/zarr-developers/zarr-python/blob/a941224ab51dfb8ca98f9a53afb91f3b6d2c6955/src/zarr/core/chunk_grids.py#L33) in zarr-python 3, so maybe we don't need to vendor it

Reply:
I'm always *very* reluctant to import private API, so I'd be pro still vendoring it.

Reply:
that works!